mage/query-modules/python/gnn-link-prediction.md

```diff
@@ -0,0 +1,375 @@
+MATCH (v4:PAPER {id: 13}), (v1:PAPER {id: 10}) CREATE (v4)-[e:CITES {}]->(v1);
+```
+
+  </TabItem>
+
+  <TabItem value="set-model-parameters">
+
+```cypher
+CALL link_prediction.set_model_parameters({target_relation: ["PAPER", "CITES", "PAPER"], node_features_property: "features", 
+split_ratio: 1.0, predictor_type: "mlp", num_epochs: 100, hidden_features_size: [256], attn_num_heads: [1]}) YIELD * RETURN *;
+```
+  
+  </TabItem>
+
+
+  <TabItem value="train">
+
+```cypher
+CALL link_prediction.train() YIELD training_results, validation_results
+RETURN training_results, validation_results;
+```
+
+  </TabItem>
+
+  <TabItem value="train-results">
+
+```plaintext
++--------------------+--------------------+--------------------+--------------------+--------------------+
+| epoch_num          | accuracy           | auc_score          | loss               | precision          |
++--------------------+--------------------+--------------------+--------------------+--------------------+
+| 17                 | 0.833              | 0.906              | 0.428              | 1.0                |
+| 18                 | 0.917              | 0.938              | 0.393              | 1.0                |
+| 19                 | 0.833              | 0.938              | 0.365              | 0.75               |
+| 20                 | 0.917              | 0.938              | 0.341              | 1.0                |
+| 21                 | 0.917              | 0.938              | 0.315              | 1.0                |
+| 22                 | 0.833              | 0.969              | 0.296              | 0.75               |
+| 23                 | 0.917              | 1.0                | 0.277              | 1.0                |
+| 24                 | 0.917              | 1.0                | 0.246              | 0.8                |
+| 25                 | 0.917              | 1.0                | 0.233              | 0.8                |
+| 26                 | 1.0                | 1.0                | 0.202              | 1.0                |
+```
+
+  </TabItem>
+
+  <TabItem value="predict"> 
+
+```cypher
+MATCH (v1:PAPER {id: 10})
+MATCH (v2:PAPER {id: 12})
+CALL link_prediction.predict(v1, v2)
+YIELD score
+RETURN score;
+```
+
+  </TabItem>
+
+  <TabItem value="predict-results">
+
+```plaintext
++-------+
+| score |
+| 0.104 |
+```
+
+  </TabItem>
+
+</Tabs>
+
+## FAQ
+
+### **Why you can get into problems with reverse edges?**
+
+As specified in one of the chapters above, **reverse edges** deserve a little bit more attention. Having a `reverse_edge` in your dataset can be a problem if they are not excluded from `message passing edges` in the prediction of its `opposite edge`(`supervision edge`). The best thing you can do is have a `directed` graph and we will automatically add reverse edges, if you specify `add_reverse_edges` in the `set_model_parameters` method, in a way that doesn't cause information flow. 
```

Comment:
Which chapter? Link with anchor

Actually, i would just lose the first sentence

Who is we? We will automatically add reverse edges?

Reply:
The module, not we.